sql/core/src/main/scala/org/apache/spark/sql/execution/SparkPlan.scala

```diff
@@ -69,7 +67,9 @@ abstract class SparkPlan extends QueryPlan[SparkPlan] with Logging with Serializ
 
   /** Overridden make copy also propagates sqlContext to copied plan. */
   override def makeCopy(newArgs: Array[AnyRef]): SparkPlan = {
-    SparkSession.setActiveSession(sqlContext.sparkSession)
+    if (sqlContext != null) {
```

Comment:
just curious, why wasn't the `makeCopy` problem discovered in the previous PR/investigation?

Reply:
Because it looks failed ahead. Once we go with `lazy` then it's discovered later (the exception message in the PR description).

Reply:
I see, thanks, I wondered because it seems a more generic issue, easier to happen, but probably we never met it as all the trials included `FileSourceScanExec` which caused an earlier failure... thanks.